File: src/Bolt.Core/Entities/Restaurant.cs
using System;
using System.Collections.Generic;
using System.Reflection.Metadata;
using System.Reflection.Metadata.Ecma335;
using Bolt.SharedKernel;
using Bolt.SharedKernel.Interfaces;
using NodaTime;

namespace Bolt.Core.Entities
{
    public class Restaurant : BaseEntity, IAggregateRoot
    {
        public string Name { get; set; }
        public string Description { get; set; }
        public User Type { get; set; }
        public string Address { get; set; }
        public decimal DeliveryRadius { get; set; }
        public string PhoneNumber { get; set; }
        public string Website { get; set; }
        public bool IsOpenedForOrders { get; set; }
        public Dictionary<DayOfWeek, OpeningHours> OpeningHours { get; set; }
        public RestaurantCategory Category { get; set; }
        public User Owner  { get; set; }
        public string ImageUrl { get; set; }
        

        public Restaurant()
        {
            
        }
    }
}

File: src/Bolt.Web/wwwroot/src/models/Restaurant.ts
import { Transform } from 'class-transformer';
import _ from 'lodash';
import { DateTime } from 'luxon';
import { DayOfWeek } from './DayOfWeek';
import { RestaurantCategory } from './RestaurantCategory';
import { User } from './User';
import { BaseEntity } from './BaseEntity';
import 'reflect-metadata';

export class Restaurant extends BaseEntity {
  @Transform(
    ({ value } : {value: {DayOfWeek: {item1: string, item2: string}}}) => (
      _.mapValues(value, (hours) => {
        return [DateTime.fromISO(hours.item1), DateTime.fromISO(hours.item2)];
      })
    ),
    { toClassOnly: true },
  )
  openingHours: {[K in DayOfWeek]: [DateTime, DateTime]} = {
    [DayOfWeek.Sunday]: [DateTime.local(), DateTime.local()],
    [DayOfWeek.Monday]: [DateTime.local(), DateTime.local()],
    [DayOfWeek.Tuesday]: [DateTime.local(), DateTime.local()],
    [DayOfWeek.Wednesday]: [DateTime.local(), DateTime.local()],
    [DayOfWeek.Thursday]: [DateTime.local(), DateTime.local()],
    [DayOfWeek.Friday]: [DateTime.local(), DateTime.local()],
    [DayOfWeek.Saturday]: [DateTime.local(), DateTime.local()],
  };

  name: string = '';

  description: string = '';

  address: string = '';

  deliveryRadius: number = 0;

  phoneNumber: string = '';

  website: string = '';

  isOpenedForOrders: boolean = true;

  imageUrl: string = '';

  owner: User = new User();

  category: RestaurantCategory = RestaurantCategory.Burger;

  constructor() {
    super();
  }
}
